docs/developer/release/README.md

```diff
@@ -0,0 +1,71 @@
+# Releasing
+
+This document describes the process of creating a release for the
+`grafana/agent` repo. A release includes release assets for everything inside
+the repository, including Grafana Agent and Grafana Agent Operator.
+
+The processes described here are for v0.24.0 and above.
+
+# Prerequisites
+
+These [Prerequisites](./prerequisites.md) should be done by the release shepherd 
+before taking any actions.
+
+# Workflows
+
+Once a release is scheduled, a release shepherd is determined. This person will be 
+responsible for ownership of the following workflows:
+
+- Release Candidate Publish
+  - [Actions] 1-4,6-7,9
+- Additional Release Candidate[s] Publish
+  - [Actions] 2-4,6-7,9
+- Stable Release Publish
+  - [Actions] 2-9
+- Patch Release Publish (latest version)
+  - [Actions] 2-6,8-9
```

Comment:
Even though it's tedious, I think we should explicitly list which topics to follow instead of giving number ranges here.

Reply:
I agree, I wasn't sold on this. I will give it another pass that way.